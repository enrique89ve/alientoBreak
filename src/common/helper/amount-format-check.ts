export default (v: string) => /^\d+(\.\d+)?$/.test(v);
