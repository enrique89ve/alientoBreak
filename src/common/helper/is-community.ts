export default (s: string) => s.match(/^hive-\d+/) !== null;
