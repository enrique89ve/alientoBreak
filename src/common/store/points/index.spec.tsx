import reducer, { initialState, resetAct, fetchAct, fetchedAct, errorAct } from "./index";

import { pointTransactionsInstance } from "../../helper/test-helper";

let state = initialState;

it("1- default state", () => {
    expect(state).toMatchSnapshot();
});

it("2- fetchAct", () => {
    state = reducer(state, fetchAct(0));
    expect(state).toMatchSnapshot();
});

it("3- fetched", () => {
    const list = [...pointTransactionsInstance];
    state = reducer(state, fetchedAct("1.000", "2.000", list));
    expect(state).toMatchSnapshot();
});

it("4- resetAct", () => {
    state = reducer(state, resetAct());
    expect(state).toMatchSnapshot();
});

it("5- fetchAct", () => {
    state = reducer(state, fetchAct(10));
    expect(state).toMatchSnapshot();
});

it("6- errorAct", () => {
    state = reducer(state, errorAct());
    expect(state).toMatchSnapshot();
});
import React from "react";
import { shallow } from "enzyme";
import EntryLink from "./index";

describe("EntryLink", () => {
    const historyMock = { push: jest.fn() };
    const makePathMock = jest.fn();
    const afterClickMock = jest.fn();

    const defaultProps = {
        history: historyMock,
        makePath: makePathMock,
        afterClick: afterClickMock,
        entry: {
            category: "test-category",
            author: "test-author",
            permlink: "test-permlink",
        },
        children: <div>Test Children</div>,
    };

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const wrapper = shallow(<EntryLink {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should handle click event", async () => {
        const wrapper = shallow(<EntryLink {...defaultProps} />);
        const instance = wrapper.instance() as EntryLink;

        const eventMock = {
            preventDefault: jest.fn(),
        };

        await (instance.clicked as unknown as (event: React.MouseEvent<HTMLElement, MouseEvent>) => Promise<void>)(eventMock);

        expect(eventMock.preventDefault).toHaveBeenCalled();
        expect(afterClickMock).toHaveBeenCalled();
        expect(makePathMock).toHaveBeenCalledWith(
            defaultProps.entry.category,
            defaultProps.entry.author,
            defaultProps.entry.permlink
        );
        expect(historyMock.push).toHaveBeenCalledWith(
            makePathMock.mock.results[0].value
        );
    });

    it("should handle click event with PartialEntry", async () => {
        const partialEntryProps = {
            ...defaultProps,
            entry: {
                ...defaultProps.entry,
                title: "test-title",
            },
        };

        const wrapper = shallow(<EntryLink {...partialEntryProps} />);
        const instance = wrapper.instance() as EntryLink;

        const eventMock = {
            preventDefault: jest.fn(),
        };

        const getPostMock = jest.fn().mockResolvedValueOnce(partialEntryProps.entry);
        jest.mock("../../helper/api", () => ({
            getPost: getPostMock,
        }));

        await instance.clicked(eventMock as unknown as React.MouseEvent<HTMLElement>);

        expect(eventMock.preventDefault).toHaveBeenCalled();
        expect(afterClickMock).toHaveBeenCalled();
        expect(getPostMock).toHaveBeenCalledWith(
            partialEntryProps.entry.author,
            partialEntryProps.entry.permlink
        );
        expect(makePathMock).toHaveBeenCalledWith(
            partialEntryProps.entry.category,
            partialEntryProps.entry.author,
            partialEntryProps.entry.permlink
        );
        expect(historyMock.push).toHaveBeenCalledWith(
            makePathMock.mock.results[0].value
        );
    });
});