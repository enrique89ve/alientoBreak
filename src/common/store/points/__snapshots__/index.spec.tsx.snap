// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "filter": 0,
  "loading": false,
  "points": "0.000",
  "transactions": Array [],
  "uPoints": "0.000",
}
`;

exports[`2- fetchAct 1`] = `
Object {
  "filter": 0,
  "loading": true,
  "points": "0.000",
  "transactions": Array [],
  "uPoints": "0.000",
}
`;

exports[`3- fetched 1`] = `
Object {
  "filter": 0,
  "loading": false,
  "points": "1.000",
  "transactions": Array [
    Object {
      "amount": "0.250",
      "created": "2020-08-17T12:52:16.737322+02:00",
      "id": 5150947,
      "memo": null,
      "receiver": null,
      "sender": null,
      "type": 10,
    },
    Object {
      "amount": "0.750",
      "created": "2020-08-17T10:01:22.094361+02:00",
      "id": 5149418,
      "memo": null,
      "receiver": null,
      "sender": null,
      "type": 120,
    },
  ],
  "uPoints": "2.000",
}
`;

exports[`4- resetAct 1`] = `
Object {
  "filter": 0,
  "loading": false,
  "points": "0.000",
  "transactions": Array [],
  "uPoints": "0.000",
}
`;

exports[`5- fetchAct 1`] = `
Object {
  "filter": 10,
  "loading": true,
  "points": "0.000",
  "transactions": Array [],
  "uPoints": "0.000",
}
`;

exports[`6- errorAct 1`] = `
Object {
  "filter": 10,
  "loading": false,
  "points": "0.000",
  "transactions": Array [],
  "uPoints": "0.000",
}
`;
